Conditional paths

kirbt opened the pull request:
Conditional Autos (WIP)
Split off fulltargetcommand into
      Auto cmd to update flywheels constantly based off distance
      Auto cmd to set pivot and heading based off distance

kirbt committed:
checker is real

kirbt committed:
checker is real

kirbt committed:
Merge branch 'dcmp' of https://github.com/robototes/Crescendo2024 into dcmp

kirbt committed:
its so over

kirbt committed:
its so over

kirbt committed:
amp auto

rror at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641): Unhandled exception: java.lang.ClassNotFoundException: frc.team2412.robot.util.auto.AutoLogic
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
        at frc.team2412.robot.Robot.robotInit(Robot.java:92)
        at edu.wpi.first.wpilibj.TimedRobot.startCompetition(TimedRobot.java:107)
        at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:365)
        at edu.wpi.first.wpilibj.RobotBase.lambda$startRobot$0(RobotBase.java:433)
        at java.base/java.lang.Thread.run(Thread.java:833)

kirbt committed:
amp auto

rror at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641): Unhandled exception: java.lang.ClassNotFoundException: frc.team2412.robot.util.auto.AutoLogic
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
        at frc.team2412.robot.Robot.robotInit(Robot.java:92)
        at edu.wpi.first.wpilibj.TimedRobot.startCompetition(TimedRobot.java:107)
        at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:365)
        at edu.wpi.first.wpilibj.RobotBase.lambda$startRobot$0(RobotBase.java:433)
        at java.base/java.lang.Thread.run(Thread.java:833)

jbko6 committed:
Merge pull request #115 from robototes/dcmp

DCMP Changes

kirbt committed:
auto !

kirbt committed:
auto !

kirbt committed:
tuning autos

kirbt committed:
tuning autos

jamesdooley4 commented on src/main/java/frc/team2412/robot/commands/launcher/AimTowardsSpeakerCommand.java:
```diff
@@ -0,0 +1,80 @@
+						.getPose()
+						.getTranslation()
+						.plus(new Translation2d(fieldSpeed.vxMetersPerSecond, fieldSpeed.vyMetersPerSecond));
+		Translation2d robotToSpeaker = SPEAKER_POSITION.minus(robotPosition);
+		double distance = robotToSpeaker.getNorm();
```
Needs to set `yawTarget`

jamesdooley4 commented on src/main/java/frc/team2412/robot/commands/launcher/AimTowardsSpeakerCommand.java:
```diff
@@ -0,0 +1,80 @@
+		double distance = robotToSpeaker.getNorm();
+		LauncherDataPoint dataPoint = LAUNCHER_DATA.get(distance);
+		launcherSubsystem.launch(dataPoint.rpm);
+		launcherSubsystem.updateDistanceEntry(distance);
```
Is this needed? Setting the flywheel speed doesn't seem like part of 'aiming'.

jamesdooley4 commented on src/main/java/frc/team2412/robot/commands/launcher/AimTowardsSpeakerCommand.java:
```diff
@@ -0,0 +1,80 @@
+						.plus(new Translation2d(fieldSpeed.vxMetersPerSecond, fieldSpeed.vyMetersPerSecond));
+		Translation2d robotToSpeaker = SPEAKER_POSITION.minus(robotPosition);
+		double distance = robotToSpeaker.getNorm();
+		LauncherDataPoint dataPoint = LAUNCHER_DATA.get(distance);
```
Should the launcher angle also be set? That seems like part of 'aiming'

jamesdooley4 commented on src/main/java/frc/team2412/robot/commands/launcher/AimTowardsSpeakerCommand.java:
```diff
@@ -0,0 +1,80 @@
+									Filesystem.getDeployDirectory().getPath(),
+									LauncherSubsystem.USE_THROUGHBORE
+											? "launcher_data_throughbore.csv"
+											: "launcher_data_lamprey.csv"));
```
This may be slow the first time it's run, which could cause a hiccup if that happens mid-match (file I/O and file parsing is generally 'slow' compared to other things). Since this is now the third place using the same file parsed into a treeemap, it'd be best to have that happen in a single place & be done before auto or teleop is started. Doing this in `robotInit()` is one place to make that happen.

jamesdooley4 commented on src/main/java/frc/team2412/robot/commands/launcher/SetSpeedSpeakerCommand.java:
```diff
@@ -0,0 +1,70 @@
+									Filesystem.getDeployDirectory().getPath(),
+									LauncherSubsystem.USE_THROUGHBORE
+											? "launcher_data_throughbore.csv"
+											: "launcher_data_lamprey.csv"));
```
See comment in AimTowardsSpeakerCommand

jamesdooley4 commented on src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java:
```diff
@@ -557,7 +559,7 @@ public void periodic() {
+		if (getPosition()
 						>= (USE_THROUGHBORE ? PIVOT_SOFTSTOP_FORWARD_THROUGHBORE : PIVOT_SOFTSTOP_FORWARD)
 								+ PIVOT_DISABLE_OFFSET
 				|| launcherAngleEncoder.getPosition()
```
Second use of `launcherAngleEncoder` also should be switched to `getPosition()`

kirbt committed:
aimtowardspeakercommand does things correctly

kirbt committed:
aimtowardspeakercommand does things correctly

kirbt committed:
launcher data only used once

kirbt committed:
launcher data only used once

jamesdooley4 commented on src/main/java/frc/team2412/robot/commands/launcher/AimTowardsSpeakerCommand.java:
```diff
@@ -75,6 +70,10 @@ public void end(boolean interrupted) {
+						drivebaseSubsystem.getPose().getRotation().getDegrees(),
+						yawTarget.getDegrees(),
+						HEADING_TOLERANCE)
+				&& launcherSubsystem.isAtSpeed());
```
What sets the target speed setpoint now? The call to `launcherSubsystem.launch(...)` was removed.

jamesdooley4 replied:
Or should this be checking `launcherSubsystem.isAtAngle(...)`, since the angle target of the launcher is being set but this doesn't wait for it to be reached.

kirbt committed:
isatangle

kirbt committed:
isatangle

kirbt committed:
new ready to launch condition

in the chase rotating on the fly in auto, will also check to make sure the robot is done rotating

kirbt committed:
new ready to launch condition

in the chase rotating on the fly in auto, will also check to make sure the robot is done rotating

kirbt committed:
wip centerline n3 n2 n1 auto

kirbt committed:
wip centerline n3 n2 n1 auto

kirbt committed:
tune

kirbt committed:
tune

kirbt committed:
spotless

kirbt committed:
spotless

kirbt committed:
a

a

kirbt committed:
removed unnessescary methods in complex auto paths

kirbt committed:
Merge branch 'worlds' into ConditionalPaths

kirbt committed:
Merge branch 'ConditionalPaths' of https://github.com/kirbt/Crescendo2024 into ConditionalPaths

kirbt committed:
Merge branch 'robototes:main' into ConditionalPaths

kirbt committed:
trap poses

kirbt committed:
Merge branch 'ConditionalPaths' of https://github.com/kirbt/Crescendo2024 into ConditionalPaths

markpete commented on src/main/deploy/pathplanner/paths/AMP L_AN1 Q_CN1.path:
```diff
@@ -0,0 +1,80 @@
+  "waypoints": [
+    {
+      "anchor": {
+<<<<<<<< HEAD:src/main/deploy/pathplanner/paths/AMP L_AN1 Q_CN1.path
```
These markers look like a bad merge.  You may need to clean this up before checking in.
Examples include lines: 6, 14, 22, 29, 36, etc.

markpete replied:
Similar across multiple files in this PR.

markpete commented on src/main/java/frc/team2412/robot/util/auto/AutoLogic.java:
```diff
@@ -417,7 +412,7 @@ public static Command subwooferLaunch() {
 
 	public static Command visionLaunch() {
-		return (LAUNCHER_ENABLED && INTAKE_ENABLED && APRILTAGS_ENABLED
+		return (LAUNCHER_ENABLED && INTAKE_ENABLED && APRILTAGS_ENABLED && DRIVEBASE_ENABLED
```
Must all of these be enabled for visionLaunch?  Do you require LAUNCHER AND INTAKE or is it at least one of them so an OR?

markpete commented on src/main/java/frc/team2412/robot/util/auto/AutoLogic.java:
```diff
@@ -469,6 +464,21 @@ public static Command setAngleSubwoofer() {
 				.withName("Auto - SetPivotSubwooferCommand");
 	}
 
+	public static Command visionLaunch2() {
```
Maybe add a comment on the difference between visionLaunch and visionLaunch2 so that others know when to use which one and why these both exist.

markpete commented on src/main/java/frc/team2412/robot/util/auto/AutoLogic.java:
```diff
@@ -469,6 +464,21 @@ public static Command setAngleSubwoofer() {
+	}
+
+	public static Command setFlyWheelSpeaker() {
+		return (LAUNCHER_ENABLED && APRILTAGS_ENABLED && DRIVEBASE_ENABLED
```
INTAKE_ENABLED not a factor in this one?
